Python/codegen.c

```diff
@@ -825,6 +825,9 @@ codegen_process_deferred_annotations(compiler *c, location loc)
     void *key = (void *)((uintptr_t)ste->ste_id + 1);
 
     // Get the first annotation location
+    if (PyList_GET_SIZE(deferred_anno) == 0) {
+        goto error;
+    }
```

Comment:
Do we think this is something that can happen, or would it be a compiler bug if it did?

If this can happen, then this error check is fine, but we need to set an exception here before `goto error`. If it's not supposed to the possible, then this should be an `assert()` rather than goto error.